### PR TITLE
Add `RemoteStorageCallback`

### PR DESCRIPTION
**Pull request by andrei-21**

### CODE REVIEW

File: src/lipalightninglib.udl [hunk 1]
```diff
@@ -27,6 +27,56 @@ callback interface RedundantStorageCallback {
...
+    "PermanentFailure",
+
+    // This value is required by UniFFI library. Do not use the value directly.
+    "UnexpectedUniFFI",
+};
+
+// Remote key-value storage. Buckets are not managed explicitly.
+callback interface RemoteStorageCallback {
+    // Check if the remote server is reachable and the client has necessary
+    // permissions. The library will likely to call this method before starting
```
Comment (danielgranhao) [inline src/lipalightninglib.udl:51]:
```suggestion
    // permissions. The library will likely call this method before starting
```

File: src/lipalightninglib.udl [hunk 2]
```diff
@@ -27,6 +27,56 @@ callback interface RedundantStorageCallback {
...
+    // a transaction.
+    // Hint: request and cache an access tocken if needed.
```
Comment (danielgranhao) [inline src/lipalightninglib.udl:53]:
```suggestion
    // Hint: request and cache an access token if needed.
```

File: src/lipalightninglib.udl [hunk 3]
```diff
@@ -27,6 +27,56 @@ callback interface RedundantStorageCallback {
...
+    boolean check_health();
+
+    // List objects in the bucket. Return empty list if there is no such bucket.
+    [Throws=CallbackError]
+    sequence<string> list_objects(string bucket);
+
+    // Check if the bucket exists and the object in the bucket.
```
Comment (danielgranhao) [inline src/lipalightninglib.udl:60]:
```suggestion
    // Check if the bucket exists and if the object is in the bucket.
```

**Review by danielgranhao — APPROVED** [review]

File: src/lipalightninglib.udl [hunk 4]
```diff
@@ -27,6 +27,56 @@ callback interface RedundantStorageCallback {
...
+    // Consider fixing the input and retrying the request.
+    "InvalidInput",
+
+    // Recoverable problem (e.g. network issue, problem with en external service).
+    // Consider retrying the request.
+    "RuntimeError",
+
+    // Unrecoverable problem (e.g. internal invariant broken).
+    // Consider suggesting the user to report the issue to the developers.
+    "PermanentFailure",
```
Comment (gcomte) [inline src/lipalightninglib.udl:42]: We already have the 3 Error types above within `LipaError`. Is there a way how we could reuse it?

File: src/lipalightninglib.udl (hunk 1 above)
Comment (gcomte) [inline src/lipalightninglib.udl]: _The library is likely to call_ ... or  _The library will likely call_

File: src/lipalightninglib.udl (hunk 2 above)
Comment (gcomte) [inline src/lipalightninglib.udl]:
```suggestion
    // Hint: Request and cache an access token if needed.
```

File: src/lipalightninglib.udl (hunk 1 above)
Comment (gcomte) [reply src/lipalightninglib.udl]: Apparently already fixed.

File: src/lipalightninglib.udl (hunk 2 above)
Comment (gcomte) [reply src/lipalightninglib.udl]: Apparently already fixed.

**Review by gcomte — APPROVED** [review]

**Pull request merged**